components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_crypto_keys.c

```diff
@@ -0,0 +1,195 @@
+    }
+
+    return PSA_SUCCESS;
+}
+
+enum tfm_plat_err_t
+tfm_plat_get_initial_attest_key(uint8_t          *key_buf,
+                                uint32_t          size,
+                                struct ecc_key_t *ecc_key,
+                                enum ecc_curve_t *curve_type)
+
+{
+    uint8_t *key_dst = NULL;
+    uint8_t *key_src;
+    uint32_t key_size;
+
+    psa_status_t crypto_ret;
+
+    uint8_t *public_key = NULL;
+    psa_key_type_t type;
+    psa_key_type_t public_type;
+    size_t bits;
+    size_t public_key_size = 0;
+    size_t public_key_length = 0;
+
+    uint32_t initial_attestation_public_x_key_size = 0;
+    uint32_t initial_attestation_public_y_key_size = 0;
+
+    const psa_key_id_t key_id = PSA_ATTESTATION_PRIVATE_KEY_ID;
+    psa_key_handle_t handle = 0;
+
+    crypto_ret = psa_crypto_init();
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_open_key(PSA_KEY_LIFETIME_PERSISTENT, key_id, &handle);
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_get_key_information(handle, &type, &bits);
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+    if (!PSA_KEY_TYPE_IS_ECC(type))
+    {
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+    public_type = PSA_KEY_TYPE_PUBLIC_KEY_OF_KEYPAIR(type);
+    public_key_size = PSA_KEY_EXPORT_MAX_SIZE(public_type, bits);
+    public_key = (uint8_t *) malloc(public_key_size);
+    if (public_key == NULL)
+    {
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    crypto_ret = psa_export_public_key(handle,
+                                       public_key, public_key_size,
+                                       &public_key_length);
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        free(public_key);
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    /* Set the EC curve type which the key belongs to */
+    crypto_ret = get_curve(type, curve_type);
+    if (crypto_ret != PSA_SUCCESS)
+    {
+        free(public_key);
+        psa_close_key(handle);
+        return TFM_PLAT_ERR_SYSTEM_ERR;
+    }
+
+    key_src = public_key;
+    key_dst  = key_buf;
+
+    /* The representation of an ECC public key is:
+    ** -The byte 0x04
+    ** - `x_P` as a `ceiling(m/8)`-byte string, big-endian
+    ** - `y_P` as a `ceiling(m/8)`-byte string, big-endian
+    ** - where m is the bit size associated with the curve
+    ** - 1 byte + 2 * point size
+    */
+    initial_attestation_public_x_key_size = ceil((public_key_length - 1) / 2);
+    initial_attestation_public_y_key_size = ceil((public_key_length - 1) / 2);
+
+    /* Copy the x-coordinate of public key to the buffer */
+    if (initial_attestation_public_x_key_size != 0)
+    {
+        key_src = key_src + ONE_BYTE;
```

Comment:
can you please write short comment explaining why +1

Reply:
Its already described 12 line before